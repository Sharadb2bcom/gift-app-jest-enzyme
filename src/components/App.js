import React, { Component } from "react";
import Gift from "./Gifts";
import { max_number } from "./helper/index";
class App extends Component {
  state = {
    gifts: []
  };
  addGift = () => {
    const { gifts } = this.state;
    const ids = this.state.gifts.map(gift => gift.id);
    const max_id = max_number(ids);
    gifts.push({ id: max_id + 1 });
    this.setState({ gifts });
  };
  removeGift = id => {
    const gifts = this.state.gifts.filter(gift => gift.id !== id);
    this.setState({ gifts });
  };
  render() {
    return (
      <div className="app">
        <h2>Gifts Giver</h2>
        <div className="gifts-list">
          {this.state.gifts.map(gift => {
            return (
              <Gift gift={gift} removeGift={this.removeGift} key={gift.id} />
            );
          })}
        </div>
        <button className="btn-add" onClick={this.addGift}>
          Add Gifts
        </button>
      </div>
    );
  }
}

export default App;
