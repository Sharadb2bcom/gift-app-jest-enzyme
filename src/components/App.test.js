import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import App from "./App";

describe("App", () => {
  const app = shallow(<App />);

  it("renders correctly ", () => {
    expect(app).toMatchSnapshot();
  });

  it("initialized the `state` of empty gifts", () => {
    expect(app.state().gifts).toEqual([]);
  });
  describe("on click of `add button`", () => {
    const id = 1;
    beforeEach(() => {
      app.find(".btn-add").simulate("click");
    });
    afterEach(() => {
      app.setState({ gifts: [] });
    });
    it("add the new gift into `state`", () => {
      expect(app.state().gifts).toEqual([{ id }]);
    });

    it("add new gift to the list", () => {
      expect(app.find(".gifts-list").children().length).toEqual(1);
    });
    it("should have the Gift component", () => {
      expect(app.find("Gifts").exists()).toEqual(true);
    });

    describe("user wants to remove the gift from state", () => {
      beforeEach(() => {
        app.instance().removeGift(id);
      });
      it("removes the gift from the state", () => {
        expect(app.state().gifts).toEqual([]);
      });
    });
  });
});
