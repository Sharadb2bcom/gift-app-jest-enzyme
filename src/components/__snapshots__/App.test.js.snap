// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Gifts Giver
        </h2>,
        <div
          className="gifts-list"
        />,
        <button
          className="btn-add"
          onClick={[Function]}
        >
          Add Gifts
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gifts Giver",
        },
        "ref": null,
        "rendered": "Gifts Giver",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "gifts-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Gifts",
          "className": "btn-add",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Gifts",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Gifts Giver
          </h2>,
          <div
            className="gifts-list"
          />,
          <button
            className="btn-add"
            onClick={[Function]}
          >
            Add Gifts
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gifts Giver",
          },
          "ref": null,
          "rendered": "Gifts Giver",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "gifts-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Gifts",
            "className": "btn-add",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Gifts",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
